packages/astro/src/build/sitemap.ts

```diff
@@ -1,17 +1,17 @@
 import type { BuildOutput } from '../@types/astro';
-
 import { canonicalURL } from './util.js';
 
 /** Construct sitemap.xml given a set of URLs */
 export function generateSitemap(buildState: BuildOutput, site: string): string {
   const uniqueURLs = new Set<string>();
 
   // TODO: find way to respect <link rel="canonical"> URLs here
-  // TODO: find way to exclude pages from sitemap
+  // TODO: find way to exclude pages from sitemap (currently only skips 404 pages)
 
   // look through built pages, only add HTML
   for (const id of Object.keys(buildState)) {
     if (buildState[id].contentType !== 'text/html') continue;
+    if (id.startsWith('404.html')) continue;
```

Comment:
I think `id === "404.html"` makes a big more sense, since html is the end of the id

Reply:
Done.